webhook_router/parse.py

```diff
@@ -131,3 +140,18 @@ def _validate_missing_keys(webhook: dict) -> ValidationResult:
                 False, f"{expected_workflow_job_key} key not found in {webhook}"
             )
     return ValidationResult(True, "")
+
+
+def _parse_labels(labels: Collection[str], ignore_labels: Collection[str]) -> Labels:
```

Comment:
`Iterable` is not `Sized', so I chose `Collection` to indicate that we need a finite Iterable for the set comprehension.

Reply:
I think it is generally implied that even if the type is iterable that it shouldn't be infinitely sized unless it is a special case. Collection is usually used in cases where the requirement is to be able to iterate multiple times or access elements at specific indexes

Reply:
just two cents from my experience with iterables -
I think iterables should be used carefully since a generator can also be an iterable, I've had cases where i exhaust the iterable because the function 1. validates the items in the iterable 2. processes the iterables. But since generator can only be traversed once, step 2 will fail.
Would there be any types that can cover such situation so that statically we can avoid the issues?

Reply:
In a generic matter, I don't necessarily agree with the statement that we must choose the most permissive type, but since I've a strong background in java which is strongly type, it's a habit I have from there. Python has a radically different philosophy in terms of typing and is way more permissive (I don't personally like this :p)
Also worth mentioning that Collection inherits Sized but also Container (which enforce __contains__), I've the feeling that this type of requirements for labels is necessary. Finally we're in a method called "parse" which means we also do some validation, so enforcing some properties on the object used in that method through typing make sense.

Reply:
Thanks sharing the opinion. I will now close this conversation in order to merge the PR, as it seems there are different views on this topic that should be discussed elsewhere. cc @jdkandersson

Reply:
let's discuss during the architecture office hours. The most permissive means the most permissive for the function. So I'm not saying the function should be flexible, instead I'm saying that if the function doesn't require a more restrictive type, then it should accept a more permissive type.

So if the function depends on the collection type, i.e., has a given length, then it should only accept at least that type. In this case, there was no dependency on a size so the function should not require that from the type it accepts.

You are right, Yanks, that an iterator will only give a value once. So if a function requires being able to iterate over the value multiple times, then either it should convert to a type that permits that or only accept types that permit that. In this case that isn't required, so it shouldn't be required from the function type